src/react-imgix.jsx

```diff
@@ -331,12 +351,32 @@ class SourceImpl extends Component {
       defaultAttributeMap,
       this.props.attributeConfig
     );
+    const setParentRef = ref => {
+      if (!ref) {
+        return;
+      }
+
+      // assign ref based on if it's a callback vs object
+      if (typeof ref === "function") {
+        ref(this.sourceRef);
+      } else {
+        ref.current = this.sourceRef;
+      }
+    };
```

Comment:
I think we could keep this code more DRY and performant by making the signature `const setParentRef = (parentRef, el) =>` so that we don't have to redefine it in every render method